Content.Shared/Xenoarchaeology/Artifact/Components/XenoArtifactComponent.cs

```diff
@@ -0,0 +1,114 @@
+
+    [ViewVariables]
+    public Container NodeContainer = default!;
+
+    /// <summary>
+    /// The nodes in this artifact that are currently "active."
+    /// This is cached and updated when nodes are removed, added, or unlocked.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public List<NetEntity> CachedActiveNodes = new();
+
+    [DataField, AutoNetworkedField]
+    public List<List<NetEntity>> CachedSegments = new();
+
+    [DataField, AutoNetworkedField]
+    public bool Suppressed;
+
+    /// <summary>
+    /// A multiplier applied to the calculated point value
+    /// to determine the monetary value of the artifact
+    /// </summary>
+    [DataField]
+    public float PriceMultiplier = 0.10f;
+
+    #region Unlocking
+    /// <summary>
+    /// How long does the unlocking state last.
+    /// </summary>
+    [DataField]
+    public TimeSpan UnlockStateDuration = TimeSpan.FromSeconds(10);
+
+    /// <summary>
+    /// Minimum waiting time between unlock states.
+    /// </summary>
+    [DataField]
+    public TimeSpan UnlockStateRefractory = TimeSpan.FromSeconds(10);
+
+    [DataField, AutoPausedField]
+    public TimeSpan NextUnlockTime;
+    #endregion
+
+    // NOTE: you should not be accessing any of these values directly. Use the methods in SharedXenoArtifactSystem.Graph
+    #region Graph
+    /// <summary>
+    /// List of all of the nodes currently on this artifact.
+    /// Indexes are used as a lookup table for <see cref="NodeAdjacencyMatrix"/>.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public NetEntity?[] NodeVertices = [];
+
+    /// <summary>
+    /// Adjacency matrix that stores connections between this artifact's nodes.
+    /// A value of "true" denotes an directed edge from node1 to node2, where the location of the vertex is (node1, node2)
+    /// A value of "false" denotes no edge.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public List<List<bool>> NodeAdjacencyMatrix = new();
+
+    public int NodeAdjacencyMatrixRows => NodeAdjacencyMatrix.Count;
+    public int NodeAdjacencyMatrixColumns => NodeAdjacencyMatrix.TryGetValue(0, out var value) ? value.Count : 0;
+    #endregion
+
+    #region GenerationInfo
+
+    /// <summary>
+    /// The total number of nodes that make up this artifact.
+    /// </summary>
+    [DataField]
+    public MinMax NodeCount = new(10, 16);
+
+    /// <summary>
+    /// The amount of nodes that go in each segment.
+    /// A segment is an interconnected series of nodes.
+    /// </summary>
+    [DataField]
+    public MinMax SegmentSize = new(5, 8);
+
+    /// <summary>
+    /// For each "layer" in a segment (set of nodes with equal depth), how many will we generate?
+    /// </summary>
+    [DataField]
+    public MinMax NodesPerSegmentLayer = new(1, 3);
+
+    [DataField]
+    public MinMax ScatterPerLayer = new(0, 2);
+
+    [DataField]
+    public ProtoId<WeightedRandomEntityPrototype> EffectWeights = "XenoArtifactEffectsDefault";
```

Comment:
todo: replace with custom type (same as used for triggers! for validation purposes!)

Reply:
Its wierd how it was done but... huh, i guess we are not doing this...